ci.sh

```diff
@@ -45,7 +45,7 @@ clone() {
 # build dmd, druntime, phobos
 build() {
     source ~/dlang/*/activate # activate host compiler
-    make -j$N -C src -f posix.mak MODEL=$MODEL HOST_DMD=$DMD ENABLE_RELEASE=1 ENABLE_WARNINGS=1 all
+    make -j$N -C src -f posix.mak MODEL=$MODEL HOST_DMD=$DMD ENABLE_RELEASE=1 ENABLE_ASSERTS=1 ENABLE_WARNINGS=1 all
```

Comment:
This should probably be manually enabled per CI by passing `ENABLE_ASSERTS=1` as an environment variable. 

(To tackle https://github.com/dlang/dmd/pull/11523/files#r471124647)

Reply:
I don't get your point, and I thought we've established that omitting `-release` is the way to go, better than to rely on host-compiler-version-specific `-check` switches. `-release` is IMO there for a reason and is the standard way of en/disabling default safety checks.

Reply:
To quote @wilzbach:

> One thing that we shouldn't forget is that -release might generate different code with DMD, so it might be worthwhile to do this only for certain CIs.

There are still releases compiled by `dmd` + `-release` IIRC, so we should test it at least on one CI. Or am I missing a CI that isn't affected by your changes?

Reply:
Oh sry I misunderstood; TBH, I couldn't imagine checked assertions etc. to have side effects wrt. generated code, but well, it most likely cannot be ruled out. I'm not too familiar with DMD's CI and what each one does exactly, so someone else pls take care of selecting the job(s) to be built without assertions.

Reply:
Me neither but better safe than sorry,

I would suggest removing this line s.t. SemaphoreCI remains in it's current state and all other CI's run without `-release`.

Reply:
Done.